### PR TITLE
Adding a new section on Lint checks

### PR DESCRIPTION
**Pull request by jmslau**
Adding a new section on Lint checks. Jake/Tor - can you please review?

### CODE REVIEW

File: interop.md [hunk 1]
```diff
@@ -391,3 +392,37 @@ public class JavaClass {
     }
 }
 ```
+# Lint Checks for Interoperability
+
+## Requirements
+* **Android Studio version required:** 3.2 Canary 10 or later
+* **Lint version required:** 26.2.0-alpha10 or later
```
Comment (tnorbye) [inline interop.md]: People don't really think of the lint version using the internal sdk library version numbers; I'd instead reference this as Gradle version required, which would be 3.2.0-alpha10 or later

File: interop.md [hunk 2]
```diff
@@ -391,3 +392,37 @@ public class JavaClass {
...
+android {
+
+    ...
+
+    lintOptions {
+        check 'Interoperability'
+    }
+}```
+
+`./gradlew lint` will now run the Kotlin Interoperability checks.
```
Comment (tnorbye) [inline interop.md]: Maybe alternatively add something like this:

If you for example want to apply all the interoperability checks *except* the nullness check (which is very verbose), you can use
check 'Interoperability'
disable 'UnknownNullness'

File: interop.md (hunk 1 above)
Comment (JakeWharton) [reply interop.md]: I wouldn't even mention anything but "3.2"

File: _changes/2018-04-16.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+date: 2018-04-16
+---
+
+**Interop**
+
+ * Change: Reordered items under the [Interop guide for Java (for Kotlin consumption)](interop.html#java-for-kotlin-consumption) to roughly reflect their relative priority.
```
Comment (JakeWharton) [inline _changes/2018-04-16.md]: I rank hard keywords above nullability so I'm not sure we want to communicate this as being a priority ordering.

Comment (jmslau) [reply _changes/2018-04-16.md]: I moved hard keywords to be the first item. The priority is really just implicit. We don't say they are in priority order in the doc.

File: interop.md (hunk 1 above)
Comment (jmslau) [reply interop.md]: For people who have an earlier version of the Canary, they may try it and it doesn't work. I don't want people to getting confused, so I think it's better to be explicit. Once we go Beta/Stable, we can update.

**Comment by JakeWharton** [comment]
Manually merged and released.

**Pull request closed without merging**